sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/TooManyParametersBase.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using System.Collections.Immutable;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class TooManyParametersBase<TSyntaxKind, TParameterListSyntax> : ParameterLoadingDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TParameterListSyntax : SyntaxNode
+    {
+        protected const string DiagnosticId = "S107";
+        protected const string MessageFormat = "{2} has {1} parameters, which is greater than the {0} authorized.";
+
+        protected abstract DiagnosticDescriptor Rule { get; }
+        protected abstract TSyntaxKind[] SyntaxKinds { get; }
+        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+        protected abstract Dictionary<TSyntaxKind, string> Mapping { get; }
```

Comment:
Mapping is a really vague name. I had to look at the code to understand what it is used for.

I'd suggest `NodeToDeclarationName` instead.